logistics_project/apps/malawi/handlers/refrigerator_malfunction.py

```diff
@@ -0,0 +1,73 @@
+from datetime import datetime
+from logistics.models import ContactRole
+from logistics_project.apps.malawi.handlers.abstract.base import FacilityUserHandler
+from logistics_project.apps.malawi.models import RefrigeratorMalfunction
+from logistics.util import config
+from rapidsms.models import Contact
+
+
+class RefrigeratorMalfunctionHandler(FacilityUserHandler):
+    keyword = "rm"
+
+    def help(self):
+        self.handle("")
+
+    def is_text_valid(self, words):
+        if len(words) == 0:
+            self.respond(config.Messages.FRIDGE_HELP)
+            return False
+
+        if words[0] not in RefrigeratorMalfunction.REASONS:
+            self.respond(config.Messages.FRIDGE_HELP_REASON, code=words[0])
+            return False
+
+        return True
+
+    def respond_to_facility_user(self, reason):
+        message = {
+            RefrigeratorMalfunction.REASON_NO_GAS: config.Messages.FRIDGE_FACILITY_NO_GAS,
+            RefrigeratorMalfunction.REASON_POWER_FAILURE: config.Messages.FRIDGE_FACILITY_POWER_FAILURE,
+            RefrigeratorMalfunction.REASON_FRIDGE_BREAKDOWN: config.Messages.FRIDGE_FACILITY_BREAKDOWN,
+        }.get(reason)
+
+        self.respond(message)
+
+    def notify_district_users(self, supply_point, reason):
+        recipients = Contact.objects.filter(
+            is_active=True,
+            supply_point__location=supply_point.location.parent,
+            role=ContactRole.objects.get(code=config.Roles.DISTRICT_SUPERVISOR)
+        )
+
+        message = {
+            RefrigeratorMalfunction.REASON_NO_GAS: config.Messages.FRIDGE_DISTRICT_NO_GAS,
+            RefrigeratorMalfunction.REASON_POWER_FAILURE: config.Messages.FRIDGE_DISTRICT_POWER_FAILURE,
+            RefrigeratorMalfunction.REASON_FRIDGE_BREAKDOWN: config.Messages.FRIDGE_DISTRICT_BREAKDOWN,
+        }.get(reason)
+
+        for recipient in recipients:
+            recipient.message(message)
+
+    def malfunction_exists(self, supply_point):
+        malfunction = RefrigeratorMalfunction.get_open_malfunction(supply_point)
+
+        if malfunction:
+            days_since_report = (datetime.utcnow() - malfunction.reported_on).days
+            self.respond(
+                config.Messages.FRIDGE_MALFUNCTION_ALREADY_REPORTED,
+                days=days_since_report
+            )
+            return True
+
+        return False
+
+    def handle(self, text):
+        supply_point = self.validate_contact()
+        if supply_point:
+            words = text.split()
+            if self.is_text_valid(words):
+                reason = words[0]
+                if not self.malfunction_exists(supply_point):
```

Comment:
the else of this should probably do something... (either respond saying it's already reported, change the reason, or something else - maybe a spec clarification?)